### ISSUE TITLE
Crystal formatter doesn't remove "crystal" from code blocks in macros

### ISSUE THREAD

**Issue by jwoertink** [crystal-lang/crystal#10316]
In Crystal 0.36.0, when running `crystal tool format src`, and code block in a comment defined with `crystal` is removed.

```
# This gets formatted
# ```crystal
# puts "hi"
# ```


# To this
# ```
# puts "hi"
# ```
```

But if the comments are in a macro loop, AND also contain some sort of dynamic macro generation, then it's ignored

```
{% for n in [1] %}
  # This doesn't get formatted, unless we remove the {{ n }}
  # ```crystal
  # puts {{ n }}
  # ```
{% end %}
```

**Comment by straight-shoota** [comment]
Yes the formatter doesn't touch such macro blocks and I don't think there's any way this could work without potentially breaking something.

**Comment by asterite** [comment]
There's no way this can be fixed. As @straight-shoota says, macros are seen as almost text for the compiler. There's no semantic to them other than those `{% ... %}` and `{{ ... }}` blocks.

**Comment by jwoertink** [comment]
Cool. Yeah, not a huge deal. I just saw that and assumed it was a bug. Feel free to close it out if it's not an issue 😄
